libvirt/helper/sshconn/sshconn.go

```diff
@@ -0,0 +1,70 @@
+package sshconn
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/mitchellh/go-homedir"
+	"golang.org/x/crypto/ssh"
+)
+
+type SSHConn struct {
+	Host    string
+	Port    int
+	User    string
+	Keypath string
+	Client  *ssh.Client
+	Session *ssh.Session
+}
+
+func (s *SSHConn) Connect() error {
+	var keypath string
+	if s.Keypath == "" {
+		homepath, err := homedir.Dir()
+		if err != nil {
+			return err
+		}
+		keypath = homepath + "/.ssh/id_rsa"
+	} else {
+		keypath = s.Keypath
+	}
+
+	key, err := ioutil.ReadFile(keypath)
+	if err != nil {
+		return err
+	}
+
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		return err
+	}
+
+	config := &ssh.ClientConfig{}
+	config.SetDefaults()
+	config.User = s.User
+	config.Auth = []ssh.AuthMethod{ssh.PublicKeys(signer)}
+	config.HostKeyCallback = ssh.InsecureIgnoreHostKey()
+
+	addr := fmt.Sprintf("%s:%d", s.Host, s.Port)
+	client, err := ssh.Dial("tcp", addr, config)
+	if err != nil {
+		return err
+	} else {
+		s.Client = client
+		return nil
+	}
+}
+
+func (s *SSHConn) CombinedOutput(cmd string) (string, error) {
+	session, err := s.Client.NewSession()
```

Comment:
* can't we keep the session alive so it can be reused for all the calls?
* I know you named the function after the underlying SSH API call but it's not very intuitive. I had to read the code carefully before I realized that you're executing a command rather than combining some output.

Reply:
- We can't keep session for reuse. A Session only accepts one call to Run, Start, Shell, Output, or CombinedOutput.  Refer to: https://godoc.org/golang.org/x/crypto/ssh#Session.Run
- Agreed to change the function name

Reply:
>     * We can't keep session for reuse.

Ah ok, strange concept of a "session" they have then. :)